lib/topological_inventory/openshift/operations/worker.rb

```diff
@@ -60,24 +60,78 @@ def order_service(payload)
           )
           logger.info("Ordering #{service_offering.name} #{service_plan.name}...Complete")
 
+          context = svc_instance_context_with_url(service_offering, service_plan, service_instance )
+          status  = provisioning_status(service_instance)
+
+          update_task(task_id, :state => "completed", :status => status, :context => context)
+        rescue StandardError => err
+          update_task(task_id, :state => "completed", :status => "error", :context => {:error => err.to_s})
+        end
+
+        def update_task(task_id, state:, status:, context:)
+          task = TopologicalInventoryApiClient::Task.new("state" => state, "status" => status, "context" => context.to_json)
+          api_client.update_task(task_id, task)
+        end
+
+        def svc_instance_context_with_url(service_offering, service_plan, service_instance)
           context = {
             :service_instance => {
               :source_id  => service_plan.source_id,
-              :source_ref => service_instance.metadata&.uid
+              :source_ref => service_instance.spec&.externalID
             }
           }
 
           reason = service_instance.status.conditions.first&.reason
           status = reason == "ProvisionedSuccessfully" ? "ok" : "error"
 
-          update_task(task_id, :state => "completed", :status => status, :context => context)
-        rescue StandardError => err
-          update_task(task_id, :state => "completed", :status => "error", :context => {:error => err.to_s})
+          if status == "ok"
+            api_link = nil
+            while api_link.nil?
+              svc_instances = svc_instance_by_source_ref(service_offering.source_id,
+                                                         service_instance.spec&.externalID)
+
+              if svc_instances.meta.count > 0
+                svc_instance_id = svc_instances.data.first&.id
```

Comment:
Could handling data and meta be compartmentalized into `svc_instance_by_source_ref` ?  Seems like it belongs there and the caller doesn't need to worry about it.